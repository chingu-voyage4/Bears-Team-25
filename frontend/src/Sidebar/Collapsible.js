import React from 'react';
import PropTypes from 'prop-types';

import { Details, Summary } from './styled';

export default class Collapsible extends React.Component {
  static propTypes = {
    id: PropTypes.string.isRequired,
    children: PropTypes.oneOfType([PropTypes.object, PropTypes.array])
      .isRequired,
    title: PropTypes.string.isRequired,
    // eslint-disable-next-line react/no-unused-prop-types
    open: PropTypes.bool.isRequired,
    expanded: PropTypes.func.isRequired,
  };
  // eslint-disable-next-line
  state = {
    open: false,
  };
  static getDerivedStateFromProps(nextProps) {
    return { open: nextProps.open };
  }
  onClick = e => {
    e.stopPropagation();
    const { open } = this.state;
    this.props.expanded(this.props.id, !open);
    this.setState({ open: !open });
  };
  render() {
    const { title, children } = this.props;
    const { open } = this.state;
    const rightArrow = String.fromCharCode(9658);
    const downArrow = String.fromCharCode(9660);
    return (
      <Details>
        <Summary onClick={this.onClick}>
          {open ? downArrow : rightArrow} {title}
        </Summary>
        {open && children}
      </Details>
    );
  }
}
