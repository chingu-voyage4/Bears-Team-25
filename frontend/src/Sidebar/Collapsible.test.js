/* eslint-disable */
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Collapsible from './Collapsible';

it('should create elements', () => {
  const d = renderer
    .create(
      <Collapsible id="" title="test title" open={false} expanded={() => {}}>
        {{ _id: '1', title: 'Some article title' }}
      </Collapsible>
    )
    .toJSON();

  expect(d).toMatchSnapshot();
});

it('should expand on click', () => {
  const expand = jest.fn();
  const comp = shallow(
    <Collapsible id="" title="test title" open={false} expanded={expand}>
      {{ _id: '1', title: 'Some article title' }}
    </Collapsible>
  );
  comp.instance().onClick({ stopPropagation: () => {} });
  expect(comp).toMatchSnapshot();
  expect(expand).toHaveBeenCalled();
  expect(comp.instance().state.open).toBe(true);
});
