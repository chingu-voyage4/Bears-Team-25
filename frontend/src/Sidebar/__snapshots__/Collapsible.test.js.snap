// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create elements 1`] = `
<div
  className="sc-bdVaJa bLWtbN"
  onClick={[Function]}
>
  <div
    className="sc-bwzfXH bpFyCA"
  >
    â–º
     
    test title
  </div>
</div>
`;
